File: DNS/TwoSum.java
import java.util.Arrays;
class TwoSum {
    public static void main(String args[]) {
        //int[] nums = {-1, 0, 1, 2, -1, -4};
        int[] nums1 = {4,2,1};
        int[] result = twoSum(nums1, 8);
        System.out.println(Arrays.toString(result));
    }
    private static int[] twoSum(int[] nums, int target) {
        Arrays.sort(nums);
       int left = 0;
       int right = nums.length -1;
       int sum = 0;
       while (left < right) {
        sum = nums[left] + nums[right];
        if (sum > target) right--;
        if (sum < target) left++;
        if (sum == target) {
            return new int[] {left, right};
        }
       }
       return new int[0];
    }
}

File: python-interview-question/two_sum.py
# Spnace :O(N) and O(1) and use enumerate.
def two_sum(nums,target):
    index ={}
    i = 0
    for num in nums:
        sum = target - num
        if sum in index:
            return index[sum], i
        else:
            index[num] = i
        i +=1

nums = [4,4,4]
target = 2

print(two_sum(nums, target))